### PR TITLE
Task3.1

### PR DESCRIPTION
**Pull request by lordkroos12**

### CODE REVIEW

File: Extensions/QueueExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
Comment (IgorKheidorov) [inline Extensions/QueueExtensions.cs]: remove

File: IQueue.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿
+namespace QueueImplementation
+{
```
Comment (IgorKheidorov) [inline IQueue.cs:3]: restrict to be used for value types only

File: Queue.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	internal class Queue<T> : IQueue<T>,ICloneable
+	{
+		private LinkedList<T> _queue = new LinkedList<T>();
+		public T Dequeue()
+		{
+			T firstItem = default(T);
+			if (IsEmpty())
+			{
+				throw new InvalidOperationException("The queue is empty!");
+			}
```
Comment (IgorKheidorov) [inline Queue.cs:12]: else not needed

File: Queue.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		}
+		public bool IsEmpty()
+		{
+			return _queue.Count == 0;
+		}
+
+		public object Clone()
+		{
+			Queue<T> clonedQueue = new Queue<T>();
+			foreach (var item in _queue)
```
Comment (IgorKheidorov) [inline Queue.cs]: ?? you make the clone but make initial queue empty, this is wrong behavior

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Please fix comments

File: Queue.cs (hunk 4 above)
Comment (lordkroos12) [reply Queue.cs]: The test in the main method display the original values of the initial queue even after the clone. Maybe I misunderstood the assignment

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**